cognite/client/_api/data_modeling/data_models.py

```diff
@@ -90,11 +90,14 @@ def __iter__(self) -> Iterator[DataModel]:
         """
         return cast(Iterator[DataModel], self())
 
-    def retrieve(self, ids: DataModelIdentifier | Sequence[DataModelIdentifier]) -> DataModelList:
+    def retrieve(
+        self, ids: DataModelIdentifier | Sequence[DataModelIdentifier], inline_views: bool = False
+    ) -> DataModelList:
```

Comment:
Let's make `DataModel` generic on the return type of `DataModel:views` ( i.e. `list[View]` or `list[ViewId]`), then add overloads here on the value of `inline_views`